lms/djangoapps/courseware/views/views.py

```diff
@@ -1824,6 +1825,61 @@ def render_xblock(request, usage_key_string, check_if_enrolled=True):
         return render_to_response('courseware/courseware-chromeless.html', context)
 
 
+@require_http_methods(["GET"])
+@ensure_valid_usage_key
+@xframe_options_exempt
+@transaction.non_atomic_requests
+def render_public_video_xblock(request, usage_key_string):
+    """
+    Returns an HttpResponse with HTML content for the Video xBlock with the given usage_key.
+    The returned HTML is a chromeless rendering of the Video xBlock (excluding content of the containing courseware).
+    """
+    view = 'public_view'
+
+    usage_key = UsageKey.from_string(usage_key_string)
+    usage_key = usage_key.replace(course_key=modulestore().fill_in_run(usage_key.course_key))
+    course_key = usage_key.course_key
+
+    # usage key block type must be `video` else raise 404
+    if usage_key.block_type != 'video':
+        raise Http404("Video not found.")
+
+    with modulestore().bulk_operations(course_key):
+        try:
+            course = get_course_by_id(course_key, 0)
+        except CourseAccessRedirect:
+            raise Http404("Course not found.")  # lint-amnesty, pylint: disable=raise-missing-from
+
+        block, _ = get_module_by_usage_id(
+            request,
+            str(course_key),
+            str(usage_key),
+            disable_staff_debug_info=True,
+            course=course,
+            will_recheck_access=False
+        )
+
+        # video must be public (`Public Access` field set to True) by course author in studio in video advanced settings
+        if not block.public_access:
+            raise Http404("Video not found.")
+
+        fragment = block.render(view, context={})
+
+        context = {
+            'fragment': fragment,
+            'course': course,
+            'disable_accordion': False,
+            'allow_iframing': True,
+            'disable_header': False,
+            'disable_footer': False,
+            'disable_window_wrap': True,
+            'edx_notes_enabled': False,
+            'is_learning_mfe': True,
+            'is_mobile_app': False,
```

Comment:
Why hard code these?

Reply:
Hardcoded these because this endpoint is very specific purpose. Also I am not sure what else we can do here rather than hardcoding the values?

Reply:
Well, I might have expected something like:
```
        'is_learning_mfe': is_request_from_learning_mfe(request),
        'is_mobile_app': is_request_from_mobile_app(request),
```
And then various bits of the template are turned on or off. I don't know much about how the end user would consume this view, so maybe hardcoding True for learning MFE and False for mobile app is the right thing, if you always expect to embed calls to this view from the learning MFE and never support this on mobile?

But we do have easy ways to get the actual answers for those values, so I was just surprised to see the hardcoding.